src/java.base/share/classes/java/io/Console.java

```diff
@@ -344,16 +344,35 @@ public Charset charset() {
         throw newUnsupportedOperationException();
     }
 
+    /**
+     * {@return {@code true} if the {@code Console} instance is a terminal}
+     * <p>
+     * If it is {@code true}, the {@code Console} instance is attached to a terminal
+     * provided by the underlying platform, typically its input and output
+     * character devices are available to the {@code Console}. Otherwise it could
+     * mean that the implementation of the {@code Console} may simulate those
+     * devices within.
```

Comment:
The implNote helps as this is a difficult method to specify in a platform independent way.

The class description uses "interactive command line" and  "will typically be connected to the keyboard and display" and maybe we could re-use that here. Here's a suggestion for the second paragraph:

"This method returns true if the console device, associated with the current Java virtual machine, is a terminal, typically an interactive command line connected to a keyboard and display."

I don't have any other comments on the implementation changes, they look fine.

Reply:
Thanks. Replaced the second paragraph with your suggested sentence.